private/model/api/behavior.go

```diff
@@ -0,0 +1,320 @@
+			switch key {
+			case "AWS_REGION":
+				region = val
+			case "AWS_ACCESS_KEY":
+				accessKeyId = val
+			case "AWS_SECRET_ACCESS_KEY":
+				secretAccessKey = val
+			case "EndPointURL":
+				endpointUrl = val
+			}
+		}
+	}
+	return fmt.Sprintf("\n\t\tRegion: aws.String(%#v),\n\t\tCredentials: credentials.NewStaticCredentials(%#v, %#v, %#v),",
+		region, accessKeyId, secretAccessKey, endpointUrl)
+}
+
+// ErrorAssertExists returns true if there is an assertion in the expect map
+// which requires error as an argument
+func (c Case) ErrorAssertExists() bool {
+	for _, assertion := range c.Expect {
+		for assertionName := range assertion {
+			if strings.Contains(assertionName, "Error") {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+//template map is defined in "eventstream.go"
+var funcMap = template.FuncMap{
+	"Map": templateMap,
+}
+
+var behaviorTestTmpl = template.Must(template.New(`behaviorTestTmpl`).Funcs(funcMap).Parse(`
+
+{{define "StashCredentials"}}
+	restoreEnv := sdktesting.StashEnv() //Stashes the current environment
+	defer restoreEnv()
+{{end}}
+
+{{define "SessionSetup"}}
+	//Starts a new session with credentials and region parsed from "defaults" in the Json file'
+	sess := session.Must(session.NewSession(&aws.Config{ {{$.testCase.ConfigurationString $.Tests}} }))
+{{end}}
+
+{{define "ResponseBuild"}}
+			r.HTTPResponse = &http.Response{
+						{{- if eq $.testCase.Response.StatusCode 0}}
+											StatusCode:200,
+						{{- else}}
+							StatusCode:{{$.testCase.Response.StatusCode}},
+						{{- end}}
+							Header: http.Header{
+										{{- range $key,$val:=$.testCase.Response.Headers}}
+											"{{$key}}":[]string{ "{{$val}}" },
+										{{- end}}	
+									},
+						{{- if eq ($.testCase.Response.BodyType) "xml"}}
+							Body: ioutil.NopCloser(bytes.NewBufferString({{printf "%q" $.testCase.Response.BodyContent}})),
+						{{- else if eq ($.testCase.Response.BodyType) "json"}}
+							Body: ioutil.NopCloser(bytes.NewBuffer(
+															func() []byte {
+																v, err := json.Marshal({{printf "%#v" $.testCase.Response.BodyContent}})
+																if err != nil {
+																	panic(err)
+																}
+																return v
+															}())),
+						{{- else}}
+							Body: ioutil.NopCloser(&bytes.Buffer{}),
+						{{- end}}
+						}
+{{end}}
+
+{{define "RequestBuild"}}
+	input := {{ $.testCase.Request.BuildInputShape $.op.InputRef }}
+
+	//Build request
+	req, resp := svc.{{$.testCase.Request.Operation}}Request(input)
+	_ = resp
+
+	MockHTTPResponseHandler := request.NamedHandler{Name: "core.SendHandler", Fn: func (r *request.Request){ 
+		{{ template "ResponseBuild" Map "testCase" $.testCase -}}	
+	}}
+	req.Handlers.Send.Swap( corehandlers.SendHandler.Name, MockHTTPResponseHandler )
+
+	err := req.Send()
+	{{- if $.testCase.ErrorAssertExists}}
+		if err == nil {
+			t.Fatal(err)
+		}
+	{{- else}}
+		if err != nil {
+			t.Fatal(err)
+		}
+	{{- end}}
```

Comment:
you could condense these if/else block to reduce the duplicate code a little bit.